tools/summonerd/src/main.rs

```diff
@@ -87,6 +105,40 @@ impl Opt {
                 };
                 Ok(())
             }
+            Command::Init {
+                storage_dir,
+                phase1_root,
+            } => {
+                let file = File::open(phase1_root)?;
+                let mut reader = BufReader::new(file);
+
+                let mut phase_1_bytes = Vec::new();
+                let mut buffer = [0; 4096 * 10]; // 40 KB chunks
+
+                loop {
+                    let bytes_read = reader.read(&mut buffer)?;
+                    if bytes_read == 0 {
+                        break;
+                    }
+                    phase_1_bytes.extend_from_slice(&buffer[..bytes_read]);
+                }
```

Comment:
```suggestion
                reader.read_to_end(&mut phase_1_bytes)?;
```

I believe this should work

Reply:
(We're already doing buffered reads via BufReader)